### PR TITLE
Prepare Colors for alpha values

### PR DESCRIPTION
**Pull request by t-wissmann**
Extend Color by a alpha value and extend the parser and printer to print
colors in the #rrggbbaa format. The alpha value is not used anywhere at
the moment, but it will come in handy for decoration drawing.

Also, this adds a new test suite file dedicated to all the type parsers
and printers.

### CODE REVIEW

File: src/decoration.h [hunk 1]
```diff
@@ -10,6 +10,7 @@
 class Client;
 class Settings;
 class DecorationScheme;
+class XConnection;
```
Comment (blarz) [inline src/decoration.h:13]: Why is this needed in this change?

**Review by blarz — CHANGES_REQUESTED** [review]
Just one tiny thing

File: src/decoration.h (hunk 1 above)
Comment (t-wissmann) [reply src/decoration.h]: Thanks! That was a mistake (I was already at the next step...). I've removed the line!

Comment (blarz) [reply src/decoration.h]: Thanks!

**Review by blarz — APPROVED** [review]
